flink-connectors/flink-connector-kafka/pom.xml

```diff
@@ -225,6 +241,23 @@ under the License.
 
 	<build>
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-enforcer-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>forbid-direct-table-planner-dependencies</id>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<configuration>
+							<!-- We need to depend on table planner because we rely on its internals -->
```

Comment:
Do we rely on planner internals in this module?

Reply:
for this and the below pom changes, we based it off a similar module `flink-connector-hive` that had working `TableITCase`s. However, Chesnay suggested a cleaner solution, so i will follow through with that instead